File: jvm-client/src/test/java/com/tomaszezula/makker/client/jvm/java/UpdateScenario.java
package com.tomaszezula.makker.client.jvm.java;

import java.nio.file.Path;
import java.util.Base64;

import static com.tomaszezula.makker.client.jvm.java.Config.*;

public class UpdateScenario {
    public static void main(String[] args) {
        fromBlueprint();
        fromEncodedBlueprint();
        fromFile();
    }

    private static void fromBlueprint() {
        makeClient.updateScenario(
                        473703,
                        updatedBlueprint
                )
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }

    private static void fromEncodedBlueprint() {
        makeClient.updateScenarioEncoded(
                        473703,
                        Base64.getEncoder().encodeToString(updatedBlueprint.getBytes())
                )
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }

    private static void fromFile() {
        makeClient.updateScenario(
                        473703,
                        Path.of(getResource("blueprint-updated.json"))
                )
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }
}

File: jvm-client/src/test/kotlin/com/tomaszezula/makker/client/jvm/examples/UpdateScenario.kt
package com.tomaszezula.makker.client.jvm.examples

import com.tomaszezula.makker.client.jvm.examples.Config.makeClient
import com.tomaszezula.makker.client.jvm.examples.Config.updatedBlueprint
import com.tomaszezula.makker.common.model.Blueprint
import com.tomaszezula.makker.common.model.Scenario
import io.kotest.common.runBlocking
import java.nio.file.Path
import java.util.*

fun main() {
    runBlocking {
        fromBlueprint()
        fromEncodedBlueprint()
        fromFile()
    }
}

private suspend fun fromBlueprint() {
    makeClient.updateScenario(
        Scenario.Id(473703),
        Blueprint.Json(updatedBlueprint)
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}

private suspend fun fromEncodedBlueprint() {
    makeClient.updateScenario(
        Scenario.Id(473703),
        Blueprint.Json(
            Base64.getEncoder().encodeToString(updatedBlueprint.toByteArray()),
            encoded = true
        )
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}

private suspend fun fromFile() {
    makeClient.updateScenario(
        Scenario.Id(473703),
        Path.of(Config.getResource("blueprint-updated.json").toURI())
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}